### PR TITLE
Map auto-tuning

### PR DESCRIPTION
**Pull request by lukastruemper**

### CODE REVIEW

File: dace/optim/cutout_tuner.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import dace
+import numpy as np
+
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from dace.optim.auto_tuner import AutoTuner
```
Comment (tbennun) [inline dace/optim/cutout_tuner.py:7]: same as above

File: dace/optim/map_permutation_tuner.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import dace
```
Comment (tbennun) [inline dace/optim/map_permutation_tuner.py:1]: missing copyright

File: samples/optimization/map_tuning.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import dace
```
Comment (tbennun) [inline samples/optimization/map_tuning.py:1]: missing copyright

File: dace/optim/cutout_tuner.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import dace
```
Comment (tbennun) [inline dace/optim/cutout_tuner.py:1]: missing copyright

File: dace/optim/auto_tuner.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+from abc import ABC, abstractmethod
```
Comment (tbennun) [inline dace/optim/auto_tuner.py:1]: also missing copyright, see guidelines (and maybe add yourself to AUTHORS while you're at it)

**Review by tbennun — CHANGES_REQUESTED** [review]

File: dace/optim/map_tiling_tuner.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+            cutout = cutout_state(state, *subgraph_nodes)
+            cutout.instrument = dace.InstrumentationType.Timer
+
+            arguments = {}
+            for cstate in cutout.nodes():
+                for dnode in cstate.data_nodes():
+                    if cutout.arrays[dnode.data].transient:
+                        continue
+
+                    arguments[dnode.data] = np.copy(dreport.get_first_version(dnode.data))
```
Comment (tbennun) [inline dace/optim/map_tiling_tuner.py:57]: why copy the data? it's not going to change, is it?

File: dace/optim/map_tiling_tuner.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            results[None] = baseline
+
+            best_choice = None
+            best_runtime = baseline
+            cutout_json = cutout.to_json()
+            for point in self.space(parent_map):
+                # TODO: Hacky? Faster than deepcopy?
+                tiled_sdfg = SDFG.from_json(cutout_json)
+                # TODO: Inefficient
+                for xform in Optimizer(tiled_sdfg).get_pattern_matches(patterns=[MapTiling], options={"tile_sizes": point}):
```
Comment (tbennun) [inline dace/optim/map_tiling_tuner.py:70]: you should probably use `apply_to` on a particular map entry node, as received from cutouts (`parent_map`)?

File: dace/optim/map_tiling_tuner.py (hunk 6 above)
Comment (lukastruemper) [reply dace/optim/map_tiling_tuner.py:57]: TypeError: Passing a numpy view (e.g., sub-array or "A.T") to DaCe programs is not allowed in order to retain analyzability. Please make a copy with "numpy.copy(...)". If you know what you are doing, you can override this error in the configuration by setting compiler.allow_view_arguments to True.

Comment (tbennun) [reply dace/optim/map_tiling_tuner.py:53]: which is why I enable view arguments in the other tuning example. maybe we should enable it here too

**Pull request merged**